### PR TITLE
[cnats] Add new port.

### LINKED ISSUE #9709: [New Port Request] <nats.c>
Library name: nats.c

Library description: nats c client

Source repository URL: https://github.com/nats-io/nats.c

Project homepage (if different from the source repository):

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by sylvain-prevost**
**Describe the pull request**
 Add nats.io C client port

- #### What does your PR fix?  
  Fixes #9709, #9182, #9743

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
All, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/cnats/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,66 @@
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO nats-io/nats.c
+    REF v3.2.0
+    SHA512 570bbd5b6ed25db17755f4cbd2df9449bf1f838450e29aaa1483c11e6131293490d302031e3039d710bbbc3563ce72fb72cd3ad2c98618977a4858a5a3f2abe3
+    HEAD_REF master
+    PATCHES
```
Comment (JonLiu1993) [inline ports/cnats/portfile.cmake:8]:
```suggestion
```

File: ports/cnats/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    endif()
+endif()
+
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/${PORT})
+
+vcpkg_fixup_pkgconfig()
+
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
```
Comment (JonLiu1993) [inline ports/cnats/portfile.cmake:62]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```

File: ports/cnats/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (JonLiu1993) [inline ports/cnats/portfile.cmake:64]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

File: ports/cnats/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+
```
Comment (JonLiu1993) [inline ports/cnats/portfile.cmake:66]:
```suggestion
```

File: ports/cnats/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+endif()
+
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/${PORT})
+
+if(VCPKG_TARGET_IS_WINDOWS)
+    if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
+        if(EXISTS "${CURRENT_PACKAGES_DIR}/share/cnats/cnats-config-debug.cmake")
+            file(READ "${CURRENT_PACKAGES_DIR}/share/cnats/cnats-config-debug.cmake" DEBUG_CONFIG)
+            string(REPLACE "\${_IMPORT_PREFIX}/debug/lib/natsd.dll" "\${_IMPORT_PREFIX}/debug/bin/natsd.dll" DEBUG_CONFIG ${DEBUG_CONFIG})
+            file(WRITE "${CURRENT_PACKAGES_DIR}/share/cnats/cnats-config-debug.cmake" "${DEBUG_CONFIG}")
```
Comment (JonLiu1993) [inline ports/cnats/portfile.cmake:62]:
```suggestion
            vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/cnats/cnats-config-debug.cmake" "\${_IMPORT_PREFIX}/debug/lib/natsd.dll" "\${_IMPORT_PREFIX}/debug/bin/natsd.dll")
```

File: ports/cnats/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        endif()
+        if(EXISTS "${CURRENT_PACKAGES_DIR}/share/cnats/cnats-config-release.cmake")
+            file(READ "${CURRENT_PACKAGES_DIR}/share/cnats/cnats-config-release.cmake" RELEASE_CONFIG)
+            string(REPLACE "\${_IMPORT_PREFIX}/lib/nats.dll" "\${_IMPORT_PREFIX}/bin/nats.dll" RELEASE_CONFIG ${RELEASE_CONFIG})
+            file(WRITE "${CURRENT_PACKAGES_DIR}/share/cnats/cnats-config-release.cmake" "${RELEASE_CONFIG}")
```
Comment (JonLiu1993) [inline ports/cnats/portfile.cmake:67]:
```suggestion
            vcpkg_replace_string(${CURRENT_PACKAGES_DIR}/share/cnats/cnats-config-release.cmake" "\${_IMPORT_PREFIX}/lib/nats.dll" "\${_IMPORT_PREFIX}/bin/nats.dll")
```

**Comment by sylvain-prevost** [comment]
I messed up cnats & libgpiod branches badly.. and I do not know how to undo that cleanly vs the PR. I'm going to close both cnats & libgpiod PRs and resubmit a clean ones. sorry for the noise.

**Pull request closed without merging**